src/stream/src/common/table/state_table.rs

```diff
@@ -1122,6 +1122,55 @@ impl<S: StateStore> StateTable<S> {
     pub fn get_vnodes(&self) -> Arc<Bitmap> {
         self.vnodes.clone()
     }
+
+    /// Returns:
+    /// true: the provided pk prefix is absent in state store.
+    /// false: the provided pk prefix may or may not be present in state store.
+    pub async fn surely_not_have(&self, pk_prefix: impl Row) -> StreamExecutorResult<bool> {
+        let prefix_serializer = self.pk_serde.prefix(pk_prefix.len());
+        let encoded_prefix = serialize_pk(&pk_prefix, &prefix_serializer);
+        let encoded_key_range = range_of_prefix(&encoded_prefix);
```

Comment:
As I last comment, we only need to pass `encoded_prefix` to `state_store` so that we can avoid encode which cause another allocation and memory copy

Reply:
Maybe we can make the passed key range to be `(Bound<Bytes>, Bound<Bytes>)`, and then turning a prefix into the key range will only cause shallow copy and no allocation and memcpy.

Reply:
The performance is not the main concern point. But I think pass prefix is directly and we only support to check whether data with this prefix exists in state-store. It is clear in  for the method `surely_not_have `.